Namespace all of a table’s columns using *

Hello,

after changing to the swift2 branch it is not possible to select all table columns from a join query using [*]:

```
let db = FLFDBManager.sharedInstance.database()!
var tyres = Table(FLFDatabaseTables.MountedTyre)
let articles = Table(FLFDatabaseTables.Article)
let brands = Table(FLFDatabaseTables.Brand)
let vehicles = Table(FLFDatabaseTables.Vehicle)
let warehouses = Table(FLFDatabaseTables.Warehouse)
        
tyres = tyres.join(JoinType.LeftOuter,
            articles, on:
                tyres[FLFMountedTyre.kItemNumber] == articles[FLFArticle.kItemNumber] &&
                tyres[FLFMountedTyre.kCompanyId] == articles[FLFArticle.kCompanyId])
tyres = tyres.join(JoinType.Inner,
            brands, on:
                articles[FLFArticle.kBrand] == brands[FLFBrand.kBrandcode])
tyres = tyres.join(JoinType.Inner,
            vehicles, on:
                tyres[FLFMountedTyre.kVehicleNo] == vehicles[FLFVehicle.kVehicleNo] &&
                tyres[FLFMountedTyre.kCompanyId] == vehicles[FLFVehicle.kCompanyId])
tyres = tyres.join(JoinType.Inner,
            warehouses, on:
                tyres[FLFMountedTyre.kWarehouseNo] == warehouses[FLFWarehouse.kWarehouseNo])
        
tyres = tyres.filter(tyres[FLFMountedTyre.kCompanyId] == companyId && vehicles[FLFVehicle.kCustomerNo] == customerNo)
let x = tyres.select(tyres[*]) // Error: Cannot invoke 'select' with an argument list of tyre '(Expression<Void>)'
```

Are we doing something wrong? According to the documentation this should work as it has been in the trunk version. If we change the [*] to any column name the compile succeeds but we don't want to add every column manually in the code. 

Thank you in advance

Confirmed this is still happening with the current Swift (2.1.1). 

Any swift geeks want to take a pass at this?  Likely some some syntax change between 1.x and 2.x. The relevant code is https://github.com/stephencelis/SQLite.swift/blob/master/Source/Typed/Query.swift, around line 113:
```swift
    public func select(star: Star) -> Self {
        return select([star(nil, nil)])
    }
```

Here's a copy-paste playground repo case:

```swift
import SQLite

let db = try! Connection()
db.trace { print($0) }

let emails = Table("emails")
let to = Expression<String>("to")
let subject = Expression<String?>("subject")

try! db.run(emails.create {t in t.column(to); t.column(subject)} )
try! db.run(emails.insert(to <- "stephen@example.com", subject <- "Hello, world!" ))

let query = emails.select(emails[*])

for row in try db.prepare(query) {
    print (row)
}
```
This gives the error:
```
playground1624.swift:13:20: error: cannot invoke 'select' 
with an argument list of type '(Expression<Void>)'
let query = emails.select(emails[*])
                   ^
playground1624.swift:13:20: note: overloads for 'select' exist with these partially 
matching parameter lists: ([Expressible]), (Star), (Expression<V>), (Expression<V?>)
let query = emails.select(emails[*])
                   ^
```

--- a/SQLite/Typed/Query.swift
+++ b/SQLite/Typed/Query.swift
@@ -50,8 +50,8 @@ extension SchemaType {
     /// - Parameter all: A list of expressions to select.
     ///
     /// - Returns: A query with the given `SELECT` clause applied.
-    public func select(column1: Expressible, _ column2: Expressible, _ more: Expressible...) -> Self {
-        return select(false, [column1, column2] + more)
+    public func select(column1: Expressible, _ more: Expressible...) -> Self {
+        return select(false, [column1] + more)
     }
 
     /// Builds a copy of the query with the `SELECT DISTINCT` clause applied.
@@ -65,8 +65,8 @@ extension SchemaType {
     /// - Parameter columns: A list of expressions to select.
     ///
     /// - Returns: A query with the given `SELECT DISTINCT` clause applied.
-    public func select(distinct column1: Expressible, _ column2: Expressible, _ more: Expressible...) -> Self {
-        return select(true, [column1, column2] + more)
+    public func select(distinct column1: Expressible, _ more: Expressible...) -> Self {
+        return select(true, [column1] + more)
     }
 
     /// Builds a copy of the query with the `SELECT` clause applied.

--- a/SQLiteTests/QueryTests.swift
+++ b/SQLiteTests/QueryTests.swift
@@ -20,6 +20,14 @@ class QueryTests : XCTestCase {
     func test_select_withExpression_compilesSelectClause() {
         AssertSQL("SELECT \"email\" FROM \"users\"", users.select(email))
     }
+    
+    func test_select_withStarExpression_compilesSelectClause() {
+        AssertSQL("SELECT * FROM \"users\"", users.select(*))
+    }
+    
+    func test_select_withNamespacedStarExpression_compilesSelectClause() {
+        AssertSQL("SELECT \"users\".* FROM \"users\"", users.select(users[*]))
+    }
 
     func test_select_withVariadicExpressions_compilesSelectClause() {
         AssertSQL("SELECT \"email\", count(*) FROM \"users\"", users.select(email, count(*)))

